### PR TITLE
Add Teams Team Report

### PR DESCRIPTION
**Pull request by airicaruss**

### CODE REVIEW

**Comment by Danielabom** [comment]
Reached out to the author to get info.

File: api-reference/beta/api/reportroot-getteamsteamcounts.md [hunk 1]
```diff
@@ -0,0 +1,184 @@
+---
+title: "reportRoot: getTeamsTeamCounts"
```
Comment (Danielabom) [inline api-reference/beta/api/reportroot-getteamsteamcounts.md:2]: Please add the method to the methods table of the respective topic,

File: api-reference/beta/api/reportroot-getteamsteamcounts.md [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+description: "Get the Teams Team count by Teams type."
```
Comment (Danielabom) [inline api-reference/beta/api/reportroot-getteamsteamcounts.md:3]: We need to make capitalization changes, I suggest:
```suggestion
description: "Get the Teams team count by teams type."
```

File: api-reference/beta/api/reportroot-getteamsteamcounts.md [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+doc_type: apiPageType
+---
+
+# reportRoot: getTeamsTeamCounts
+
+Namespace: microsoft.graph
+
+[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
+
+Get the Teams Team count by Teams type.
```
Comment (Danielabom) [inline api-reference/beta/api/reportroot-getteamsteamcounts.md:16]: We need to make capitalization changes, I suggest:
```suggestion
Get the Teams team count by team type.
```

File: api-reference/beta/api/reportroot-getteamsteamcounts.md [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+GET /reports/getTeamsTeamCounts(period='{period_value}')
+```
+
+## Function parameters
+
+In the request URL, provide the following parameters with a valid value.
+
+| Parameter | Type   | Description                              |
+| :-------- | :----- | :--------------------------------------- |
+| period    | string | Specifies the length of time over which the report is aggregated. The supported values for {period_value} are: `D7`, `D30`, `D90`, and `D180`. These values follow the format D*n* where *n* represents the number of days over which the report is aggregated. Required. |
```
Comment (Danielabom) [inline api-reference/beta/api/reportroot-getteamsteamcounts.md:43]:
```suggestion
| period    | string | Specifies the length of time over which the report is aggregated. The supported values for {period_value} are: `D7`, `D30`, `D90`, and `D180`. These values follow the format D*n*, where *n* represents the number of days over which the report is aggregated. Required. |
```

File: api-reference/beta/api/reportroot-getteamsteamcounts.md [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+
+| Name          | Description               |
+| :------------ | :------------------------ |
+| Authorization | Bearer {token}. Required. |
+
+## Response
+
+### CSV
+
+If successful, this method returns a `302 Found` response that redirects to a preauthenticated download URL for the report. That URL can be found in the `Location` header in the response.
```
Comment (Danielabom) [inline api-reference/beta/api/reportroot-getteamsteamcounts.md:60]:
```suggestion
If successful, this method returns a `302 Found` response that redirects to a pre-authenticated download URL for the report. That URL can be found in the `Location` header in the response.
```

File: api-reference/beta/api/reportroot-getteamsteamcounts.md [hunk 6]
```diff
@@ -0,0 +1,184 @@
...
+| :------------ | :------------------------ |
+| Authorization | Bearer {token}. Required. |
+
+## Response
+
+### CSV
+
+If successful, this method returns a `302 Found` response that redirects to a pre-authenticated download URL for the report. That URL can be found in the `Location` header in the response.
+
+Preauthenticated download URLs are only valid for a short period of time (a few minutes) and do not require an `Authorization` header.
```
Comment (Danielabom) [inline api-reference/beta/api/reportroot-getteamsteamcounts.md:62]:
```suggestion
Pre-authenticated download URLs are only valid for a short period (a few minutes) and do not require an `Authorization` header.
```

File: changelog/Microsoft.O365Reporting.json [hunk 7]
```diff
@@ -1917,6 +1917,24 @@
...
 		"SubArea": "Microsoft 365 usage reports"
+	},
+	{
+		"ChangeList": [
+			{
+			  "Id": "e2eed40d-a104-44bd-9300-7e010ac589aa",
+			  "ApiChange": "Method",
+			  "ChangedApiName": "getTeamsTeamCounts",
+			  "ChangeType": "Addition",
+			  "Description": "Added the [getTeamsTeamCounts](https://learn.microsoft.com/graph/api/reportroot-getTeamsTeamCounts?view=graph-rest-beta) method.",
```
Comment (Danielabom) [inline changelog/Microsoft.O365Reporting.json:1928]: The description must link the method to its resource. Please see example #6 here: [https://dev.azure.com/msazure/One/_wiki/wikis/Microsoft%20Graph%20Partners/211742/Changelog]

**Review by Danielabom — CHANGES_REQUESTED** [review]
Hi @airicaruss, thank you for working on this PR. I added some comments; please review them. 
Thanks!

File: api-reference/beta/api/reportroot-getteamsteamcounts.md (hunk 1 above)
Comment (airicaruss) [reply api-reference/beta/api/reportroot-getteamsteamcounts.md:2]: Added the function to the table

File: changelog/Microsoft.O365Reporting.json (hunk 7 above)
Comment (airicaruss) [reply changelog/Microsoft.O365Reporting.json]: Added the description

File: api-reference/beta/api/reportroot-getteamsteamcounts.md (hunk 2 above)
Comment (airicaruss) [reply api-reference/beta/api/reportroot-getteamsteamcounts.md]: Fixed

File: api-reference/beta/api/reportroot-getteamsteamcounts.md (hunk 3 above)
Comment (airicaruss) [reply api-reference/beta/api/reportroot-getteamsteamcounts.md]: Fixed

**Comment by airicaruss** [comment]
@Danielabom Thank you for your comments. I have fixed the comments. Please let me know if you have any other comments

File: changelog/Microsoft.O365Reporting.json [hunk 8]
```diff
@@ -1917,6 +1917,24 @@
...
 		"SubArea": "Microsoft 365 usage reports"
+	},
+	{
+		"ChangeList": [
+			{
+			  "Id": "e2eed40d-a104-44bd-9300-7e010ac589aa",
+			  "ApiChange": "Method",
+			  "ChangedApiName": "getTeamsTeamCounts",
+			  "ChangeType": "Addition",
+			  "Description": "Added the [getTeamsTeamCounts](https://learn.microsoft.com/graph/api/reportroot-getTeamsTeamCounts?view=graph-rest-beta) method to [MicrosoftTeamsTeamActivityReport](https://learn.microsoft.com/en-us/graph/api/resources/microsoft-teams-team-usage-reports?view=graph-rest-beta).",
```
Comment (Danielabom) [inline changelog/Microsoft.O365Reporting.json:1928]:
```suggestion
			  "Description": "Added the [getTeamsTeamCounts](https://learn.microsoft.com/graph/api/reportroot-getTeamsTeamCounts?view=graph-rest-beta) method to the [MicrosoftTeamsTeamActivityReport](https://learn.microsoft.com/en-us/graph/api/resources/microsoft-teams-team-usage-reports?view=graph-rest-beta) resource.",
```

**Review by Danielabom — APPROVED** [review]
@airicaruss, thanks for the quick follow-up. PR approved.

File: api-reference/beta/api/reportroot-getteamsteamcounts.md [hunk 9]
```diff
@@ -0,0 +1,184 @@
...
+
+This method supports the `$format` [OData query parameter](/graph/query-parameters) to customize the response. The default output type is `text/csv`. However, if you want to specify the output type, you can use the OData `$format` query parameter to set the default output to `text/csv` or `application/json`.
+
+## Request headers
+
+| Name          | Description               |
+| :------------ | :------------------------ |
+| Authorization | Bearer {token}. Required. |
+
+## Response
```
Comment (Lauragra) [inline api-reference/beta/api/reportroot-getteamsteamcounts.md:56]: We are missing the required H2 section "Request body". API topics adhere to a defined structure so that users can easily find the information they're looking for. Please add the missing section.

File: api-reference/beta/api/reportroot-getteamsteamcounts.md [hunk 10]
```diff
@@ -0,0 +1,184 @@
...
+doc_type: apiPageType
+---
+
+# reportRoot: getTeamsTeamCounts
+
+Namespace: microsoft.graph
+
+[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
+
+Get the Teams team count by team type.
```
Comment (Lauragra) [inline api-reference/beta/api/reportroot-getteamsteamcounts.md:16]: I suggest that we reword this description to avoid using the word "team" three times and help clarify this statement a bit. For example, would this suggested revision be accurate: 

"Get the number of teams of a particular type in an instance of Microsoft Teams."

Also, how do they specify the team "type"?

File: api-reference/beta/toc.yml [hunk 11]
```diff
@@ -22930,6 +22930,8 @@ items:
             href: api/reportroot-getteamsteamactivitycounts.md
           - name: Get team activity distribution counts
             href: api/reportroot-getteamsteamactivitydistributioncounts.md
+          - name: Get team team counts
```
Comment (Lauragra) [inline api-reference/beta/toc.yml:22933]: I think that we can shorten this TOC entry to "Get team counts".

File: changelog/Microsoft.O365Reporting.json [hunk 12]
```diff
@@ -1917,6 +1917,24 @@
...
 		"SubArea": "Microsoft 365 usage reports"
+	},
+	{
+		"ChangeList": [
+			{
+			  "Id": "e2eed40d-a104-44bd-9300-7e010ac589aa",
+			  "ApiChange": "Method",
+			  "ChangedApiName": "getTeamsTeamCounts",
+			  "ChangeType": "Addition",
+			  "Description": "Added the [getTeamsTeamCounts](https://learn.microsoft.com/graph/api/reportroot-getTeamsTeamCounts?view=graph-rest-beta) method to the [MicrosoftTeamsTeamActivityReport](https://learn.microsoft.com/en-us/graph/api/resources/microsoft-teams-team-usage-reports?view=graph-rest-beta) resource.",
```
Comment (Lauragra) [inline changelog/Microsoft.O365Reporting.json:1928]: Is that an actual resource name? It appears to be linking to what looks like a concept topic/anchor for that section of the TOC. Please clarify this changelog entry.

**Comment by Lauragra** [comment]
@airicaruss , please see my comments on this PR and address them; this is currently blocking publishing. Thanks!

**Comment by Lauragra** [comment]
@airicaruss , FYI, I've removed the ready to merge label. When you get a chance to address the comments, please set ready to merge again. Thanks.

File: api-reference/beta/api/reportroot-getteamsteamcounts.md (hunk 10 above)
Comment (airicaruss) [reply api-reference/beta/api/reportroot-getteamsteamcounts.md:16]: The type could be 'Private Teams', 'Active Private Teams', 'Public Teams', 'Active Public Teams'

File: api-reference/beta/api/reportroot-getteamsteamcounts.md (hunk 9 above)
Comment (airicaruss) [reply api-reference/beta/api/reportroot-getteamsteamcounts.md:60]: Added

File: api-reference/beta/api/reportroot-getteamsteamcounts.md (hunk 10 above)
Comment (Lauragra) [reply api-reference/beta/api/reportroot-getteamsteamcounts.md]: @airicaruss , where do we describe this in the docs? Is there a place that we can link to?

**Comment by airicaruss** [comment]
@Lauragra could you please let me know if there is any other change required from my side?

File: api-reference/beta/api/reportroot-getteamsteamcounts.md [hunk 13]
```diff
@@ -0,0 +1,188 @@
...
+doc_type: apiPageType
+---
+
+# reportRoot: getTeamsTeamCounts
+
+Namespace: microsoft.graph
+
+[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
+
+Get the number of teams of a particular type in an instance of Microsoft Teams.
```
Comment (Lauragra) [inline api-reference/beta/api/reportroot-getteamsteamcounts.md:16]: Can you provide a link here to where we list the applicable types of teams?

File: changelog/Microsoft.O365Reporting.json [hunk 14]
```diff
@@ -1917,6 +1917,24 @@
...
 		"SubArea": "Microsoft 365 usage reports"
+	},
+	{
+		"ChangeList": [
+			{
+			  "Id": "e2eed40d-a104-44bd-9300-7e010ac589aa",
+			  "ApiChange": "Method",
+			  "ChangedApiName": "getTeamsTeamCounts",
+			  "ChangeType": "Addition",
+			  "Description": "Added the [getTeamsTeamCounts](https://learn.microsoft.com/graph/api/reportroot-getTeamsTeamCounts?view=graph-rest-beta) method to the [MicrosoftTeamsTeamActivityReport](https://learn.microsoft.com/en-us/graph/api/resources/report?view=graph-rest-beta) resource.",
```
Comment (Lauragra) [inline changelog/Microsoft.O365Reporting.json:1928]: The link to MicrosoftTeamsTeamActivityReport that we're calling a resource is not a resource. It's an overview topic. Please clarify this changelog entry.

**Comment by Danielabom** [comment]
@airicaruss, Laura Graham added some feedback. Please review it. Thanks!

**Comment by airicaruss** [comment]
> @airicaruss, Laura Graham added some feedback. Please review it. Thanks!

@Danielabom could you please help with my comment?

File: changelog/Microsoft.O365Reporting.json [hunk 15]
```diff
@@ -1917,6 +1917,24 @@
...
 		"SubArea": "Microsoft 365 usage reports"
+	},
+	{
+		"ChangeList": [
+			{
+			  "Id": "e2eed40d-a104-44bd-9300-7e010ac589aa",
+			  "ApiChange": "Method",
+			  "ChangedApiName": "getTeamsTeamCounts",
+			  "ChangeType": "Addition",
+			  "Description": "Added the [getTeamsTeamCounts](https://learn.microsoft.com/graph/api/reportroot-getTeamsTeamCounts?view=graph-rest-beta) method to the **MicrosoftTeamsTeamActivityReport** resource.",
```
Comment (Lauragra) [inline changelog/Microsoft.O365Reporting.json:1928]: Please verify the name of the resource here - the binding parameter in the CSDL is reportRoot and I'm not finding a resource named MicrosoftTeamsTeamActivityReport. I think we have an issue with the reportRoot resource type topic not linking to the functions that have the reportRoot binding type. Also, the reportRoot resource topic is not currently surfaced in the TOC. @airicaruss  and @Danielabom , can you please sort this out? Thanks.

**Comment by airicaruss** [comment]
Hi @Lauragra , could you please help to review and comment if there is any further changes required?

**Comment by Lauragra** [comment]
> Hi @Lauragra , could you please help to review and comment if there is any further changes required?

Hi @airicaruss , I had one remaining comment about clarifying/linking to the types of teams. I'm not sure what that means. Do you think it will be clear to your customers? If you don't plan to fix, we can approve and publish this PR. Thanks.

**Comment by airicaruss** [comment]
> out clarifying/lin

Is there any 

> > Hi @Lauragra , could you please help to review and comment if there is any further changes required?
> 
> Hi @airicaruss , I had one remaining comment about clarifying/linking to the types of teams. I'm not sure what that means. Do you think it will be clear to your customers? If you don't plan to fix, we can approve and publish this PR. Thanks.

Is there any suggestion if we describe it in another way? @Zhiliqiao888 need your confirm if this description is good for publish.

**Comment by airicaruss** [comment]
> to

@Lauragra  synced internally and we would like to get this doc checked in first. We don't have good idea to modify the description now. We will optimize the doc later if customer found it confusing. Could you please help to approve and merge this change?

**Review by Lauragra — APPROVED** [review]
